modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TasksPostExecutionHandler.java

```diff
@@ -174,4 +197,35 @@ private int getPossibleErrorsNumber() {
     private boolean isRetryScheduled(Map<String, Object> metadata) {
         return metadata.get(TASK_RETRY) != null && (boolean) metadata.get(TASK_RETRY);
     }
+
+
+    private boolean shouldScheduleRetry(Task task, Map<String, Object> metadata, int retryNumber) {
+        Map<String, String> taskRetries = new HashMap<>();
+        List<Integer> retryInterval = new ArrayList<>();
+        int numberOfRetries;
+
+        try {
+            InputStream retries = settings.getRawConfig(TASK_PROPERTIES_FILE_NAME);
+            Properties props = new Properties();
+            props.load(retries);
+            if (props != null) {
+                for (Map.Entry<Object, Object> entry : props.entrySet()) {
+                    taskRetries.put((String) entry.getKey(), (String) entry.getValue());
+                    retryInterval.add(0, Integer.valueOf(entry.getValue().toString()));
+                }
+            }
+            numberOfRetries = taskRetries.size();
+        } catch (IOException e) {
+            throw new MotechException("Error loading raw file config to properties", e);
+        }
+
+        if(retryNumber < numberOfRetries) {
+            task.setRetryIntervalInMilliseconds(retryInterval.get(retryNumber) * 1000);
+        } else {
+            metadata.put(TASK_RETRY, true);
+        }
+
+        return metadata.get(TASK_RETRY) != null && (boolean) metadata.get(TASK_RETRY);
```

Comment:
it is difficult to understand what is going on here, please make it more readable